backend/src/models/analysis.py

```diff
@@ -106,3 +106,25 @@ def units_to_annotate(self):
                     })
 
         return units
+
+    def find_discussion_post(self, discussion_post_id):
+        """ Finds a specific discussion post in an analysis by the discussion post id """
```

Comment:
No change needed, but in the future, its helpful to include that the method will return None if there is no discussion with that post_id in this analysis.

Reply:
This description has been updated to note that None will be returned if no post is found matching the id.